skrub/_join_utils.py

```diff
@@ -218,3 +222,95 @@ def _get_new_name(suggested_name, forbidden_names):
         return suggested_name
     token = _utils.random_string()
     return f"{untagged_name}__skrub_{token}__"
+
+
+def left_join(left, right, left_on, right_on, rename_right_cols="{}"):
+    """Left join two dataframes of the same type.
+
+    The input dataframes type must agree: either both `left` and `right` need to be
+    :obj:`pandas.DataFrame`, :obj:`polars.DataFrame` or :obj:`polars.LazyFrame`.
+    Mixing types will raise an error.
+
+    `rename_right_cols` can be used to format the right dataframe columns, e.g. use
+    "right_.{}" to rename all right cols with a leading "right_".
+
+    If duplicate column names are found between renamed right cols and left cols,
+    a __skrub_<random string>__ is added at the end of columns that would otherwise
+    be duplicates.
+
+    Parameters
+    ----------
+    left : pd.DataFrame or pl.DataFrame or pl.LazyFrame
+        The left dataframe of the left-join.
+    right : pd.DataFrame or pl.DataFrame or pl.LazyFrame
+        The right dataframe of the left-join.
+    left_on : str or Iterable[str]
+        Left keys to merge on.
+    right_on : str or Iterable[str]
+        Right keys to merge on.
+    rename_right_cols : str, optional
+        Formatting used to rename right cols, by default "{}".
+
+    Returns
+    -------
+    pd.DataFrame or pl.DataFrame or pl.LazyFrame
```

Comment:
maybe just "dataframe"? Let's try to limit the number of places where we explicitly list the possible dataframe libraries so we have less to edit if we ever add Ibis

Reply:
Keeping all the types only once in the docstring so that it's clear, replacing other occurrences with "dataframe"